packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -284,6 +285,25 @@ export async function runHttpQuery(
         );
       }
 
+      if (optionsObject.dataSources) {
+        const dataSources = optionsObject.dataSources();
+
+        const httpCache = new HTTPCache();
+
+        for (const dataSource of Object.values(dataSources)) {
+          dataSource.httpCache = httpCache;
```

Comment:
The other option would be to provide them as arguments to the dataSources api. Also I'm worried that these methods are called for each request, so with the current setup, it doesn't appear that there would be any caching, since the HttpCache is created each time